packages/core/src/motify.tsx

```diff
@@ -1,32 +1,30 @@
-import React, { ComponentType, forwardRef } from 'react'
-import type { ImageStyle, TextStyle, ViewStyle } from 'react-native'
+import React, {
+  Component,
+  ComponentType,
+  forwardRef,
+  PropsWithChildren,
+} from 'react'
+import type { ImageProps, StyleProp, TextProps, ViewProps } from 'react-native'
 import type { MotiProps } from './types'
 import useMapAnimateToStyle from './use-map-animate-to-style'
 import Animated from 'react-native-reanimated'
 
 // https://www.framer.com/blog/posts/magic-motion/
 export default function motify<
-  Style,
-  Props extends { style?: Style },
-  Ref,
-  ExtraProps,
-  // Variants,
-  // Animate = ViewStyle & TextStyle
-  Animate = ViewStyle | ImageStyle | TextStyle
+  Props extends ViewProps | TextProps | ImageProps,
```

Comment:
This seems to restrict the component to require one of these props. This won't work for custom components. That's why I originally made the style optional. Maybe something like `Pick<ViewProps | TextProps | ImageProps, 'style'>`? I want to make sure style isn't restricted. It might even make sense to make `Props extends object`.

Reply:
Ahh yeah, you're right! This seems to extend things while keeping type inference:

```ts
Props extends { style?: object | StyleProp<any> },
```

Reply:
That looks like it would make sense

Reply:
Just checking in on updating this, thanks.